hypertrace-gradle-publish-plugin/src/main/java/org/hypertrace/gradle/publishing/PublishPlugin.java

```diff
@@ -58,9 +58,12 @@ private void addPublishRepository() {
   }
 
   private void addKnownPublications() {
-    validateGradlePropertiesBeforePublishTask();
-    getPublishingExtension().publications(this::addJavaLibraryPublicationWhenApplied);
-    getPublishingExtension().publications(this::addDistributionPublicationWhenApplied);
+    project.afterEvaluate(unused -> {
```

Comment:
the problem with moving this all into an afterEvaluate block is actually the inverse of what we had - now we know the script will have already been run, but the script won't be able to override anything we run. That's why I keep pushing for lazy config :) We want our config to be _declared_ before the script runs but _evaluated_ after.

The new check we added, for example, need not run until we actually try to access license.

Reply:
without `afterEvaluate`, values from new extension are not available.

Reply:
Right - that's why we bind the references to them without explicit `.get()` calls (see https://github.com/hypertrace/hypertrace-gradle-publish-plugin/pull/19#discussion_r608603438 )

Reply:
removed `afterEvaluate`. Added license check before `generatePomFileForJavaLibraryPublication` task.